velox/exec/Window.cpp

```diff
@@ -350,10 +386,50 @@ void Window::callApplyForPartitionRows(
     rawPeerEnds[j] = peerEndRow_ - 1 - firstPartitionRow;
   }
 
-  auto updateFrameBounds = [&](vector_size_t* rawFrameBounds,
+  auto readFrameValues = [&](const FrameChannelArg& frameArg) -> void {
+    frameArg.value->resize(numRows);
+    windowPartition_->extractColumn(
+        frameArg.index, partitionOffset_, numRows, 0, frameArg.value);
+  };
+
+  auto columnValuesCheck = [&](const int64_t* offsets) -> void {
```

Comment:
This variable name is too generic. Let's use a regular helper function instead of a lambda and write a simple loop instead of std::all_of and make sure to include invalid values in the error message. Also, make sure to generate USER error, not SYSTEM error.